### PR TITLE
improve cookie parsing

### LINKED ISSUE #1118: Cookie.CookieError: Illegal key value: _csrf/link
##### Steps to reproduce the problem:

1. mitmdump -t .
2. visit http://dailymotion.com using the proxy
3. website tries to set cookie _csrf/link, which crashes mitmdump

##### What is the expected behavior?

I would expect mitmdump to not crash, but either forward the cookie as is, or not forward it.

##### What went wrong?

According to RFC 6265 (Cookie RFC), which refers to RFC 2616 (HTTP), "_csrf/link" is an invalid
cookie name because it contains a "/". Visiting the website with Chrome does not segfault Chrome, but instead continues as if the cookiename is valid. I believe mitmdump should handle this error gracefully instead of crashing.

Traceback (most recent call last):
  File "/usr/local/bin/mitmdump", line 9, in <module>
    load_entry_point('mitmproxy==0.18', 'console_scripts', 'mitmdump')()
  File "/usr/local/lib/python2.7/dist-packages/mitmproxy/main.py", line 102, in mitmdump
    master.run()
  File "/usr/local/lib/python2.7/dist-packages/mitmproxy/dump.py", line 350, in run
    super(DumpMaster, self).run()
  File "/usr/local/lib/python2.7/dist-packages/mitmproxy/controller.py", line 27, in run
    self.tick(0.1)
  File "/usr/local/lib/python2.7/dist-packages/mitmproxy/flow.py", line 845, in tick
    return super(FlowMaster, self).tick(timeout)
  File "/usr/local/lib/python2.7/dist-packages/mitmproxy/controller.py", line 39, in tick
    handle_func(obj)
  File "/usr/local/lib/python2.7/dist-packages/mitmproxy/dump.py", line 334, in handle_response
    flow.FlowMaster.handle_response(self, f)
  File "/usr/local/lib/python2.7/dist-packages/mitmproxy/flow.py", line 1039, in handle_response
    self.process_new_response(f)
  File "/usr/local/lib/python2.7/dist-packages/mitmproxy/flow.py", line 936, in process_new_response
    self.stickycookie_state.handle_response(f)
  File "/usr/local/lib/python2.7/dist-packages/mitmproxy/flow.py", line 337, in handle_response
    c = http_cookies.SimpleCookie(str(i))
  File "/usr/lib/python2.7/Cookie.py", line 583, in __init__
    if input: self.load(input)
  File "/usr/lib/python2.7/Cookie.py", line 640, in load
    self.__ParseString(rawdata)
  File "/usr/lib/python2.7/Cookie.py", line 677, in __ParseString
    self.__set(K, rval, cval)
  File "/usr/lib/python2.7/Cookie.py", line 589, in __set
    M.set(key, real_value, coded_value)
  File "/usr/lib/python2.7/Cookie.py", line 459, in set
    raise CookieError("Illegal key value: %s" % key)
Cookie.CookieError: Illegal key value: _csrf/link

I would submit a fix myself, but I'm a bit swamped with other work right now

##### Any other comments?

---

Mitmproxy Version: 0.18
Operating System: Ubuntu

### PR DESCRIPTION
**Pull request by Kriechi**
allows `/` to be within a cookie name, e.g. `_csrf/link=foobar`
backport: `:` character to be within a cookie name from python 3.3
removes deprecated cookie getter/setter
fixes #1118 

This introduces a minor API change for sticky cookies.

### CODE REVIEW

File: netlib/http/cookies.py [hunk 1]
```diff
@@ -27,6 +28,12 @@
 
 # TODO: Disallow LHS-only Cookie values
 
+class BrowserCookie(Cookie.BaseCookie):
```
Comment (mhils) [inline netlib/http/cookies.py]: Add a docstring here why we do this dance?

File: test/netlib/http/test_response.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ class TestResponseUtils(object):
     def test_get_cookies_none(self):
         resp = tresp()
         resp.headers = Headers()
-        assert not resp.cookies
+        assert len(resp.cookies) == 0
```
Comment (mhils) [inline test/netlib/http/test_response.py]: I'd actually say that `not resp.cookies` looks *a bit*  more pythonic, but I don't have a strong opinion on this. :relaxed:

**Comment by mhils** [comment]
LGTM!

File: netlib/http/cookies.py [hunk 3]
```diff
@@ -203,25 +203,26 @@ def refresh_set_cookie_header(c, delta):
...
-                # reddit.com sends a an expires tag like this: "Thu, 31 Dec
-                # 2037 23:59:59 GMT", which is valid RFC 1123, but not
-                # strictly correct according to the cookie spec. Browsers
-                # appear to parse this tolerantly - maybe we should too.
-                # For now, we just ignore this.
-                del i["expires"]
-    ret = c.output(header="").strip()
+
+    if "expires" in attrs:
+        e = parsedate_tz(attrs["expires"][0])
```
Comment (mhils) [inline netlib/http/cookies.py]: We use the first one here, whereas we use the last one at https://github.com/mitmproxy/mitmproxy/pull/1121/files#diff-6d576d016bf3f8c7c5ff10b0b3ee4bfdR323 - is any of those ways more correct?

(We should never trigger this anyway, so both is fine for me - just curious)

**Comment by mhils** [comment]
LGTM! :+1:

File: netlib/http/cookies.py (hunk 3 above)
Comment (Kriechi) [inline netlib/http/cookies.py]: I'm going with the last one - assuming each occurrence overwrites an existing value.

**Pull request merged**